sys/fmt/fmt.c

```diff
@@ -137,6 +137,10 @@ uint32_t scn_u32_dec(const char *str, size_t n)
 
 void print(const char *s, size_t n)
 {
+#ifdef __WITH_AVRLIBC__
+    /* AVR's libc doesn't offer write(), so use fwrite() instead */
+    fwrite(s, n, 1, stdout);
```

Comment:
Thanks for the explanation. Makes sense to me and I guess even if the AVR's serial port fails, looping and retrying won't do any good here.